File: tests/fixtures/dotnet_solution/src/Api/Controllers/UsersController.cs
using Microsoft.AspNetCore.Mvc;
using Acme.Domain;

namespace Acme.Api;

[ApiController]
[Route("api/users")]
public class UsersController : ControllerBase
{
    private readonly IUserRepository _repo;
    public UsersController(IUserRepository repo) { _repo = repo; }

    [HttpGet("{email}")]
    public async Task<IActionResult> Get(string email)
    {
        var user = await _repo.FindAsync(email);
        return user is null ? NotFound() : Ok(user);
    }

    [HttpPost]
    public async Task<IActionResult> Create([FromBody] User user)
    {
        await _repo.AddAsync(user);
        return Created($"/api/users/{user.Email}", user);
    }
}

File: tests/fixtures/vbnet_solution/src/Api/UsersController.vb
Imports Microsoft.AspNetCore.Mvc
Imports Acme.Domain

<ApiController>
<Route("api/users")>
Public Class UsersController
    Inherits ControllerBase

    Private ReadOnly _repo As IUserRepository

    Public Sub New(repo As IUserRepository)
        _repo = repo
    End Sub

    <HttpGet("{email}")>
    Public Async Function [Get](email As String) As Task(Of IActionResult)
        Dim user = Await _repo.FindAsync(email)
        Return If(user Is Nothing, NotFound(), CType(Ok(user), IActionResult))
    End Function

    <HttpPost>
    Public Async Function Create(<FromBody> user As User) As Task(Of IActionResult)
        Await _repo.AddAsync(user)
        Return Created($"/api/users/{user.Email}", user)
    End Function
End Class
